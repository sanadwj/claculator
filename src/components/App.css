body {
  margin: 0 auto;
}

.calculator {
  display: flex;
  flex-direction: column;
  max-width: 700px;
}

.display {
  display: flex;
  max-width: 700px;
  background-color: gray;
  height: 100px;
  color: white;
  font-weight: bold;
  padding: 20px;
  justify-content: flex-end;
  align-items: center;
  margin: 0;
}

.buttonPanel {
  max-width: 700px;
}

.buttonDiv {
  width: 100%;
}
