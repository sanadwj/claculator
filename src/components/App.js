import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import NavBar from './NavBar';
import Calculator from './Calculator';
import Home from './Home';
import Quote from './Quote';

const App = () => (

  <>
    <div>

      <Router>
        <NavBar />
        <Switch>
          <Route exact path="/calculator" component={Calculator} />
          <Route exact path="/" component={Home} />
          <Route exact path="/quote" component={Quote} />
        </Switch>
      </Router>
    </div>
  </>

);

export default App;
