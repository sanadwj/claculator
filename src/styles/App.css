body {
  margin: 0 auto;
}

.calculator {
  display: flex;
  flex-direction: column;
  width: 700px;
  padding: 30px;
}

.display {
  display: flex;
  max-width: 700px;
  background-color: gray;
  height: 100px;
  color: white;
  font-weight: bold;
  padding: 20px;
  justify-content: flex-end;
  align-items: center;
  margin: 0;
}

.buttonPanel {
  max-width: 700px;
}

.buttonDiv {
  width: 100%;
}

.navbar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 20px;
}

.navbar h1 {
  font-size: 45px;
}

.navbar ul {
  display: flex;
}

.navbar ul li {
  font-weight: bold;
  padding: 5px;
  text-decoration: none;
  list-style-type: none;
}

.quote {
  display: block;
  justify-content: center;
  align-items: start;
  padding: 50px;
}

.home {
  padding: 20px;
}

.lorem {
  padding: 30px;
}

.calculate {
  display: flex;
  justify-content: space-between;
}

.calculate h2 {
  padding: 20px;
}
