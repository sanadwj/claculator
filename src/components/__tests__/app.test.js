import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

import App from '../App';

describe('App', () => {
  it('renders', () => {
    const tree = renderer.create(
      <Router><App /></Router>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
