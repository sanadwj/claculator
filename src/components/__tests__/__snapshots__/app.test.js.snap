// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<div>
  <nav
    className="navbar"
  >
    <h1>
      Math Magicians
    </h1>
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home |
        </a>
      </li>
      <li>
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator |
        </a>
      </li>
      <li>
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
  <h2
    className="home"
  >
    Welcome to our page!
  </h2>
  <p
    className="lorem"
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab aperiam architecto aspernatur atque commodi earum exercitationem expedita facere hic incidunt itaque laborum magnam minima molestiae molestias praesentium quas quibusdam recusand.
  </p>
</div>
`;

exports[`Display renders correctly 1`] = `
<div>
  <nav
    className="navbar"
  >
    <h1>
      Math Magicians
    </h1>
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home |
        </a>
      </li>
      <li>
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator |
        </a>
      </li>
      <li>
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
  <h2
    className="home"
  >
    Welcome to our page!
  </h2>
  <p
    className="lorem"
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab aperiam architecto aspernatur atque commodi earum exercitationem expedita facere hic incidunt itaque laborum magnam minima molestiae molestias praesentium quas quibusdam recusand.
  </p>
</div>
`;
